src/Send/SendScreen/SendScreen.tsx

```diff
@@ -64,34 +62,37 @@ export const SendScreen = ({
   const strings = useStrings()
   const navigation = useNavigation()
   const wallet = useSelectedWallet()
-
-  const tokenBalance = useSelector(tokenBalanceSelector)
   const isFetchingBalance = useSelector(isFetchingUtxosSelector)
   const lastFetchingError = useSelector(lastUtxosFetchErrorSelector)
   const defaultAsset = getDefaultAssetByNetworkId(wallet.networkId)
+  const balance = useBalances(wallet, defaultAsset.identifier)
   const utxos = useSelector(utxosSelector)
   const hasPendingOutgoingTransaction = useSelector(hasPendingOutgoingTransactionSelector)
   const netInfo = useNetInfo()
   const isOnline = netInfo.type !== 'none' && netInfo.type !== 'unknown'
-  const selectedAsset = tokenBalance.values.find(({identifier}) => identifier === selectedTokenIdentifier)
 
-  if (!selectedAsset) {
+  const selectedAssetAvailableAmount = balance[selectedTokenIdentifier]
+
+  if (!selectedAssetAvailableAmount) {
     throw new Error('Invalid token')
   }
 
-  const [address, setAddress] = React.useState('')
-  const [addressErrors, setAddressErrors] = React.useState<AddressValidationErrors>({addressIsRequired: true})
-  const [amountErrors, setAmountErrors] = React.useState<AmountValidationErrors>({amountIsRequired: true})
-  const [balanceErrors, setBalanceErrors] = React.useState<BalanceValidationErrors>({})
-  const [balanceAfter, setBalanceAfter] = React.useState<BigNumber | null>(null)
-  const [yoroiUnsignedTx, setYoroiUnsignedTx] = React.useState<null | YoroiUnsignedTx>(null)
-  const [fee, setFee] = React.useState<BigNumber | null>(null)
-  const [recomputing, setRecomputing] = React.useState(false)
-  const [showSendAllWarning, setShowSendAllWarning] = React.useState(false)
+  const defaultAssetAvailableAmount = balance[defaultAsset.identifier]
+
+  const [address, setAddress] = useState('')
```

Comment:
what this a required change?

Reply:
What is the preference to do it this way or the other way using `React` directly?

Reply:
lint rule could be cool

Reply:
yes, but those types of changes should be in a separate pr

Reply:
Next time I won't mix

Reply:
how about this time you dont mix

Reply:
Reverted